### DISCUSSION TITLE
reset + defaultChecked doesn't set the values for checkbox group

### DISCUSSION THREAD

**Discussion by PraveenPeri** [react-hook-form/react-hook-form#5443] [Q&A] [answered]
**Describe the bug**
Material's checkbox group doesn't have the initial state correctly set through defaultChecked when the initial values are coming from reset(). I don't think this problem is specific to Material-UI though.

**To Reproduce**

1. Create a checkbox group using Material-UI
2. Use reset to provide the initial checked states of the boxes
3. Use defaultChecked to set the initial checked state
4. See that the checked state is not correctly set

**Codesandbox link (Required)**
https://codesandbox.io/s/patient-sunset-pt5jz

**Expected behavior**
Should be able to set the initial checkbox state using the values from reset. (These values are coming from an API in our application).

**Desktop (please complete the following information):**

- OS: Linux
- Browser: Edge
- Version: 90.0.818.66

**Comment by bluebill1049** [comment]
<img width="1156" alt="Screen Shot 2021-06-02 at 8 29 23 am" src="https://user-images.githubusercontent.com/10513364/120397962-a403b700-c37c-11eb-83bb-8e5fe6a76485.png">

**Comment by PraveenPeri** [comment]
Could you try it now?

**Comment by bluebill1049** [comment, marked as answer]
This is not a bug. https://codesandbox.io/s/blue-cache-ye96r?file=/src/App.js default value is working, now please go fix this part, I need to start my job.
<img width="769" alt="Screen Shot 2021-06-02 at 9 20 46 am" src="https://user-images.githubusercontent.com/10513364/120401688-d1079800-c383-11eb-9961-ce756a06c1cb.png">

**Reply by PraveenPeri** [reply]
My bad! Looks like I did not complete save my sandbox. Try this again - https://codesandbox.io/s/patient-sunset-pt5jz

**Reply by bluebill1049** [reply]
did you see my codesandbox above?

**Reply by PraveenPeri** [reply]
I did. I could not use defaultValues but only rely on reset() because the initial values are coming from an API to which, the client makes an ajax call after the initial render of the form. If you look at my current sandbox, it took out defaultValues completely, tweaked the handleCheck() method and trying to set initial check state through reset.

**Reply by PraveenPeri** [reply]
Nevermind! defaultValue prop on the Controller and changing defaultChecked to checked on the Checkbox component did the trick. I set it to an empty array so that the initial render goes through just fine and reset() will provide array of actual values later.
Thanks again! And sorry for bothering
